compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -323,18 +324,30 @@ void addPackPeel4LevelTilingBasedPassPipeline(
   funcPassManager.addPass(createCanonicalizerPass());
   funcPassManager.addPass(createCSEPass());
 
-  // First level packing
-  {
-    AMDAIEPackAndTransposeOptions packOptions;
-    packOptions.packLevel = 0;
-    funcPassManager.addPass(createAMDAIEPackAndTransposePass(packOptions));
+  // First level pack or pad operation depending on the number of input loops.
```

Comment:
Could you add a similar comment as above about mmt4d etc:
> // The initial number of loops from the linalg ops determines whether the
  // first level of packing is needed. If the number of loops is larger
  // than 4, it means the input is not a standard 2D matmul, then there is no
  // need to pack operands from 2D to 4D.